### ISSUE TITLE
False negative with `@Nullable` annotation on type parameter in `extends`

### ISSUE THREAD

**Issue by msridhar** [jspecify/nullness-checker-for-checker-framework#223]
```java
import org.jspecify.annotations.*;
class Test {
    interface Supplier<T extends @Nullable Object> {}
    interface SingleSupplier<T> extends Supplier<@Nullable T> {}
    
    static SingleSupplier<Object> make() { throw new RuntimeException(); }
    Supplier<Object> supplier = make();
}
```
I get no error for this example with the reference checker, but I think I should at the final assignment, since `SingleSupplier<Object>` is a subtype of `Supplier<@Nullable Object>`, which is not a subtype of `Supplier<Object>`.  Please let me know if my understanding is incorrect; thanks!

**Comment by msridhar** [comment]
I do get an error with EISOP for a similar example.

FYI @cpovirk @netdpb

**Comment by cpovirk** [comment]
I agree that there ought to be an error.

**Comment by netdpb** [comment]
This is all in null-unmarked context, right? Does that make the lack of error less suprising?

**Comment by msridhar** [comment]
Whoops, I forgot the `@NullMarked`!  🤦‍♂️ With that annotation added I do get an error.  Sorry for the noise!
